lib/Runtime/Library/ArrayBuffer.cpp

```diff
@@ -1042,68 +1047,85 @@ namespace Js
         }
         uint32 growSize = newBufferLength - this->bufferLength;
 
-        bool failedReport = false;
-        const auto reportFailedFn = [&failedReport] { failedReport = true; };
+        struct AutoFailFastOnBadState
+        {
+            bool isStateValid = true;
+            ~AutoFailFastOnBadState()
```

Comment:
QueryContinue can throw javascript exception. In that case we fail fast?
Check AutoDisableInterrupt in threadcontext.h, which is suited for this need.